tests/integration/targets/ecs_cluster/tasks/20_ecs_service.yml

```diff
@@ -0,0 +1,905 @@
+    placement_strategy:
+      - type: binpack
+        field: MEMORY
+    desired_count: 1
+    state: present
+  register: ecs_service_creation_strategy
+
+- name: Assert ecs service strategy
+  assert:
+    that:
+      - ecs_service_creation_strategy.changed
+      - "ecs_service_creation_strategy.service.placementStrategy | length == 1"
+      - "ecs_service_creation_strategy.service.placementStrategy[0].type == 'binpack'"
+      - "ecs_service_creation_strategy.service.placementStrategy[0].field == 'MEMORY'"
+
+- name: Update ecs service's placement strategy
+  vars:
+    ansible_python_interpreter: "{{ botocore_virtualenv_interpreter }}"
+  ecs_service:
+    name: "{{ ecs_service_name }}-strategy"
+    cluster: "{{ ecs_cluster_name }}"
+    force_new_deployment: true
+    load_balancers:
+      - targetGroupArn: "{{ elb_target_group_instance.target_group_arn }}"
+        containerName: "{{ ecs_task_name }}"
+        containerPort: "{{ ecs_task_container_port }}"
+    task_definition: "{{ ecs_task_name }}:{{ ecs_task_definition.taskdefinition.revision }}"
+    scheduling_strategy: "REPLICA"
+    placement_strategy:
+      - type: spread
+        field: instanceId
+    desired_count: 1
+    state: present
+  register: ecs_service_update_strategy
+
+- name: Assert ecs service strategy
+  assert:
+    that:
+      - ecs_service_update_strategy.changed
+      - "ecs_service_update_strategy.service.placementStrategy | length == 1"
+      - "ecs_service_update_strategy.service.placementStrategy[0].type == 'spread'"
+      - "ecs_service_update_strategy.service.placementStrategy[0].field == 'instanceId'"
+
+- name: Remove ecs service's placement strategy
+  vars:
+    ansible_python_interpreter: "{{ botocore_virtualenv_interpreter }}"
+  ecs_service:
+    name: "{{ ecs_service_name }}-strategy"
+    cluster: "{{ ecs_cluster_name }}"
+    force_new_deployment: true
+    load_balancers:
+      - targetGroupArn: "{{ elb_target_group_instance.target_group_arn }}"
+        containerName: "{{ ecs_task_name }}"
+        containerPort: "{{ ecs_task_container_port }}"
+    task_definition: "{{ ecs_task_name }}:{{ ecs_task_definition.taskdefinition.revision }}"
+    scheduling_strategy: "REPLICA"
+    purge_placement_strategy: true
+    desired_count: 1
+    state: present
+  register: ecs_service_remove_strategy
+
+- name: Assert ecs service strategy
+  assert:
+    that:
+      - ecs_service_remove_strategy.changed
+      - "ecs_service_remove_strategy.service.placementStrategy | length == 0"
+# ============================================================
+# Begin tests for Fargate
+
+- name: ensure AmazonECSTaskExecutionRolePolicy exists
+  iam_role:
+    name: "{{ ecs_task_role_name }}"
+    assume_role_policy_document: "{{ lookup('file','ecs-trust-policy.json') }}"
+    description: "Allows ECS containers to make calls to ECR"
+    state: present
+    create_instance_profile: false
+    managed_policy:
+    - AmazonEC2ContainerServiceRole
```

Comment:
I feel like this probably needs to be `AmazonECSTaskExecutionRolePolicy`. The `AmazonEC2ContainerServiceRole` seems to be phased out in favor of the service linked role. Unfortunately, even if I change this, I'm still having problems with permissions.